Compiler/src/Preprocessor.hs

```diff
@@ -56,32 +95,41 @@ preprocessCond (BinOp op a b) = do
     res <- expandCond $ BinOp op newA newB
     pure $ newAllocA ++ newAllocB ++ res
 
-expandNegate :: Precondition -> Precondition
-expandNegate (BinOp OpEq a b) = BinOp OpNe a b
-expandNegate (BinOp OpNe a b) = BinOp OpEq a b
-expandNegate (BinOp OpLt a b) = BinOp OpGe a b
-expandNegate (BinOp OpGt a b) = BinOp OpLe a b
-expandNegate (BinOp OpLe a b) = BinOp OpGt a b
-expandNegate (BinOp OpGe a b) = BinOp OpLt a b
-expandNegate (BinOp OpIn a b) = BinOp OpNotIn a b
-expandNegate (BinOp OpNotIn a b) = BinOp OpIn a b
-expandNegate (Conj conds) = Disj $ map expandNegate conds
-expandNegate (Disj conds) = Conj $ map expandNegate conds
-expandNegate (NegateCond cond) = cond
-
-allocateVar :: Locator -> State Env ([Stmt], Locator)
+expandNegate :: Precondition Preprocessing -> State (Env Typechecking) (Precondition Preprocessing)
```

Comment:
Does this need to be in the state monad? It doesn't seem to access any of the state.